### ISSUE TITLE
"use_sudo_password" leaves "pyinfra-sudo-askpass" executable file on host after deploy finishes

### ISSUE THREAD

**Issue by taranlu-houzz** [Fizzadar/pyinfra#543]
**Describe the bug**
It seems that `use_sudo_askpass` does not clean up after itself. There aren't any real security issues caused by this, since the file is benign (although, maybe someone could inject a payload by modifying that file if it exists if pyinfra doesn't regenerate it each time?). It's more of an issue of not leaving clutter.

**To Reproduce**
Run a deploy that uses `use_sudo_askpass`.

Not sure if maybe this is only an issue on macOS.

**Expected behavior**
See above.

**Meta**
```
❯ pyinfra --support
--> Support information:

    If you are having issues with pyinfra or wish to make feature requests, please
    check out the GitHub issues at https://github.com/Fizzadar/pyinfra/issues .
    When adding an issue, be sure to include the following:

    System: Darwin
      Platform: macOS-10.14.6-x86_64-i386-64bit
      Release: 18.7.0
      Machine: x86_64
    pyinfra: v1.3.7
    Executable: /Users/----/.virtualenvs/----/bin/pyinfra
    Python: 3.9.2 (CPython, Clang 11.0.0 (clang-1100.0.33.17))
```

**Comment by Fizzadar** [comment]
I've seen this before - it affects everywhere! Need to add a hook to remove it if created.

**Comment by Fizzadar** [comment]
This is now fixed in https://github.com/Fizzadar/pyinfra/commit/3377c54789240b383a3e2c7fa4e396f10eb896fb and released in `v1.3.8`!
